### DISCUSSION TITLE
Imported sets - default to hidden from students?

### DISCUSSION THREAD

**Discussion by taniwallach** [openwebwork/webwork2#1651] [Ideas]
At present imported sets are immediately considered "published" and visible to students. I think that is a bad default.

Maybe we can either provide an option to make sets "hidden" immediately on import or just make that the default behavior.

**Comment by Alex-Jordan** [comment]
While I agree in principle, there is a problem in practice with making the default be unpublished. It would add one more step for explaining to "casual" WeBWorK using faculty about what they must do to get going. Which is already a headache. I think because of this I'd prefer to have unpublished be an option. Of course I am biased because at my school there are so many "casual" users.

**Comment by dlglin** [comment]
I think that both can be done without too much difficulty.  Have a checkbox or dropdown on the import page that allows instructors to decide whether imported sets should be published or not, and have a config option that allows you to set the default choice.
